webui/src/components/editrecord.jsx

```diff
@@ -567,6 +567,9 @@ const EditRecord = React.createClass({
         const afterPatch = (record) => {
             if (this.props.isDraft && !this.isForPublication()) {
                 this.props.refreshCache();
+                // TODO(edima): when a draft is publised, clean the state of
```

Comment:
The suggestion in the note imo is better than what we have now, where we clear the cache completely before each fetch.